codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/S3AddPutObjectUnseekableBodyDoc.java

```diff
@@ -0,0 +1,88 @@
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+import software.amazon.smithy.codegen.core.CodegenException;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.traits.DocumentationTrait;
+import software.amazon.smithy.utils.MapUtils;
+import software.amazon.smithy.utils.Pair;
+import software.amazon.smithy.utils.SetUtils;
+
+public class S3AddPutObjectUnseekableBodyDoc implements GoIntegration {
+    private static final Logger LOGGER = Logger.getLogger(S3AddPutObjectUnseekableBodyDoc.class.getName());
+
+    private static final Map<ShapeId, Set<Pair<ShapeId, String>>> SERVICE_TO_SHAPE_MAP = MapUtils.of(
+            ShapeId.from("com.amazonaws.s3#AmazonS3"), SetUtils.of(
+                    new Pair(ShapeId.from("com.amazonaws.s3#PutObjectRequest"), "Body"),
+                    new Pair(ShapeId.from("com.amazonaws.s3#UploadPartRequest"), "Body")
+            )
+    );
+
+    @Override
+    public byte getOrder() {
+        // This integration should happen before other integrations that rely on the presence of this trait
+        return -60;
```

Comment:
Wouldn't pre-process model always run prior to usage of model? Don't think this ordering is required.

Reply:
Can remove. The ordering is always present, will inherit from parent if not defined here. But yeah I don't think order maters in this case. Other customizations where order model is pre-processed is where this will be relevant.

Reply:
>  Other customizations where order model is pre-processed is where this will be relevant.

Can't think of a use case where order of pre-processing documentation would matter.